### PR TITLE
DOCSP-32094 Add more query examples

### PR DESCRIPTION
**Pull request by jeff-allen-mongo**
## DESCRIPTION

Added two new examples to the Compass query bar page:

- Match on multiple conditions ($and)
- Match with $or.

As part of these updates, I reformatted the query bar examples to not use tabs. This way, the example headings appear in the right-hand toc and act as jump-list so users can quickly find examples. The new examples are noted in the in-line comments.

## STAGING

https://docs-mongodbcom-staging.corp.mongodb.com/compass/docsworker-xlarge/DOCSP-32094/query/filter/#examples

## JIRA

https://jira.mongodb.org/browse/DOCSP-32094

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=64da4b5b67dbe910108ee119

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Is this free of spelling errors?
- [x] Is this free of grammatical errors?
- [x] Is this free of staging / rendering issues?
- [x] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/query/filter.txt [hunk 1]
```diff
@@ -47,6 +47,251 @@ Set Query Filter
...
+   {
+      "_id": { "$oid": "5e349915cebae490877d561d" },
+      "name": "Andrea Le",
+      "email": "andrea_le@fake-mail.com",
+      "version": 5,
+      "scores": [ 85, 95, 75 ],
+      "dateCreated": { "$date": "2003-03-26" }
+   }
+
+Match Documents by Multiple Conditions
```
Comment (jeff-allen-mongo) [inline source/query/filter.txt:128]: This is the newly-added example.

File: source/query/filter.txt [hunk 2]
```diff
@@ -47,6 +47,284 @@ Set Query Filter
...
+   {
+      "_id": { "$oid": "5e349915cebae490877d561d" },
+      "name": "Andrea Le",
+      "email": "andrea_le@fake-mail.com",
+      "version": 5,
+      "scores": [ 85, 95, 75 ],
+      "dateCreated": { "$date": "2003-03-26" }
+   }
+
+Match Documents by Multiple Conditions ($and)
```
Comment (jeff-allen-mongo) [inline source/query/filter.txt:128]: New example.

File: source/query/filter.txt [hunk 3]
```diff
@@ -47,6 +47,284 @@ Set Query Filter
...
+   {
+      "_id": { "$oid":"5a9427648b0beebeb69579cf" },
+      "name": "Greg Powell",
+      "email": "greg_powell@fake-mail.com",
+      "version": 1,
+      "scores": [ 65, 75, 80 ],
+      "dateCreated": { "$date": "1999-02-10" }
+   }
+
+Match Documents by Multiple Possible Conditions ($or)
```
Comment (jeff-allen-mongo) [inline source/query/filter.txt:152]: New example.

File: source/query/filter.txt [hunk 4]
```diff
@@ -47,6 +47,283 @@ Set Query Filter
...
+
+.. code-block:: JSON
+   :copyable: false
+
+   [
+      {
+         "_id": { "$oid":"5e349915cebae490877d561e" },
+         "email": "no_name@fake-mail.com",
+         "version": 4,
+         "scores": [ 90, 90, 75 ],
```
Comment (jmd-mongo) [inline source/query/filter.txt:206]: [typo]

The value for this field should be [ 90, 90, 70 ] according to the inserted documents.

File: source/query/filter.txt [hunk 5]
```diff
@@ -47,6 +47,283 @@ Set Query Filter
...
```
Comment (jmd-mongo) [inline source/query/filter.txt:245]: Similar comment as on line 206.

File: source/query/filter.txt [hunk 6]
```diff
@@ -47,6 +47,283 @@ Set Query Filter
...
+         "email": "andrea_le@fake-mail.com",
+         "version": 5,
+         "scores": [ 85, 95, 75 ],
+         "dateCreated": { "$date": "2003-03-26" }
+      },
+      {
+         "_id": { "$oid": "5e349915cebae490877d561e" },
+         "email": "no_name@fake-mail.com",
+         "version": 4,
+         "scores": [ 90, 90, 75 ],
```
Comment (jmd-mongo) [inline source/query/filter.txt:292]: Similar comment as on line 206.

File: source/query/filter.txt [hunk 7]
```diff
@@ -47,6 +47,283 @@ Set Query Filter
 
    .. include:: /includes/extracts/query-bar-results.rst
 
+Examples
```
Comment (jmd-mongo) [inline source/query/filter.txt:50]: [praise]

I think this much more scannable than the previous version and the right-hand nav titles are a clear win. 👍

File: source/query/filter.txt [hunk 8]
```diff
@@ -47,6 +47,283 @@ Set Query Filter
...
+--------
+
+The examples on this page use a small example dataset. To import the
+sample data into your MongoDB deployment, perform the following steps:
+
+1. Copy the following documents to your clipboard:
```
Comment (jmd-mongo) [inline source/query/filter.txt:56]: [question]

I'm curious if there's any benefit to using a `procedure::` directive as opposed to using an ordered list here. 

Either way seems reasonable, I just wanted to note it.

**Review by jmd-mongo — COMMENTED** [review]
Thanks for the improvements and updates here, @jeff-allen-mongo! 

I noted some small issues and have also left a question for your consideration.

Thanks!
Joe

File: source/query/filter.txt (hunk 8 above)
Comment (jeff-allen-mongo) [reply source/query/filter.txt:56]: Good idea. I'll update to use the procedure directive.

File: source/query/filter.txt (hunk 4 above)
Comment (jeff-allen-mongo) [reply source/query/filter.txt]: Fixed.

File: source/query/filter.txt (hunk 5 above)
Comment (jeff-allen-mongo) [reply source/query/filter.txt]: Fixed.

File: source/query/filter.txt (hunk 6 above)
Comment (jeff-allen-mongo) [reply source/query/filter.txt]: Fixed.

File: source/query/filter.txt (hunk 8 above)
Comment (jeff-allen-mongo) [reply source/query/filter.txt:56]: Ah, so using the procedure directive caused the step titles to appear in the right-hand toc, which results in a confusing / messy presentation.

I think if the steps didn't appear in the toc, the procedure directive would improve the presentation of the main page, but I don't think it works well in this case. Thanks for the suggestion!

Comment (jmd-mongo) [reply source/query/filter.txt:56]: Got it -- thanks for the info!

**Review by jmd-mongo — APPROVED** [review]
Thanks for the updates and giving the `procedure::` directive a shot. This one LGTM!

Thanks,
Joe

**Pull request merged**